File: src/types/merchants.ts
import type { Syncable } from './common'

/** 商户（Merchant，ADR-0028）：交易付款对象/收入来源的参考数据字典行。
 *  名字字典（issue #223）：icon/color 已退役，只保留名称。 */
export interface Merchant extends Syncable {
  id: string
  name: string
  /** 软删标记：仅随含软删全量列表（includeDeleted=true）返回软删行为 true（issue #191） */
  is_deleted: boolean
}

export interface MerchantInput {
  name: string
}

/** 更新入参：name 可省略（省略即保持原值）；改名须避开在用同名。 */
export interface MerchantUpdateInput {
  name?: string
}

File: src-tauri/src/models/merchants.rs
//! 商户领域模型：商户实体与入参（issue #188 / ADR-0028）。
//!
//! 商户是参考数据字典（与分类/账户同款模式）：`name` 在用行全库唯一、
//! 软删除；交易以 `merchant_id` 引用（见核心交易域 Transaction），
//! 改名/软删经命令面收敛（见 `commands::merchants`）。
//! 商户回归「名字字典」：`icon` / `color` 已退役（issue #223）。

use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

use crate::db::query::FromRow;

#[derive(Debug, Serialize, Deserialize, Clone, ToSchema)]
pub struct Merchant {
    pub id: String,
    pub name: String,
    pub created_at: String,
    pub updated_at: String,
    pub version: i64,
    pub device_id: String,
    pub is_deleted: bool,
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct MerchantInput {
    pub name: String,
}

/// 更新入参：`name` 可省略（省略即保持原值，等价空更新）；改名须避开在用同名。
#[derive(Debug, Deserialize)]
pub struct MerchantUpdateInput {
    pub name: Option<String>,
}

impl FromRow for Merchant {
    fn from_row(row: &rusqlite::Row) -> rusqlite::Result<Self> {
        Ok(Merchant {
            id: row.get(0)?,
            name: row.get(1)?,
            created_at: row.get(2)?,
            updated_at: row.get(3)?,
            version: row.get(4)?,
            device_id: row.get(5)?,
            is_deleted: row.get::<_, i64>(6)? != 0,
        })
    }
}
